typing/types.ml

```diff
@@ -232,13 +232,12 @@ type type_declaration =
     type_expansion_scope: int;
     type_loc: Location.t;
     type_attributes: Parsetree.attributes;
-    type_immediate: Type_immediacy.t;
     type_unboxed: unboxed_status;
     type_uid: Uid.t;
  }
 
 and type_kind =
-    Type_abstract
+    Type_abstract of {immediate: Type_immediacy.t}
```

Comment:
I would prefer having a proper record than an inline record:
- this is destined to grow if we apply your approach to more things currently stored in signatures
- I could see functions taking this information as argument, so it deserves a name

Reply:
I'm generally unenthusiastic about adding things because we might maybe want them someday. It's a very easy change to make after the fact, let's wait until someone actually needs it and let them do it then, with a specific use in mind.